problems/http.go

```diff
@@ -21,8 +22,18 @@ func Generic(status int) Problem {
 func WriteResponse(ctx context.Context, err error, w http.ResponseWriter, r *http.Request) {
 	problem := FromError(err)
 
-	if decoratableProblem, ok := problem.(RequestDecoratableProblem); ok {
+	// reflect.Value to the underlying implementation of the Problem interface
```

Comment:
I think you can remove the comments.
They are basically describing what the following line does. I think.


for sure we can remove "DecorateWithRequest should have changed the problem" meaning previous line should have side effect.

Reply:
Yes the comments are more for the next one looking at this code. For me it was not 100% clear why it was needed in the first place. Maybe just have a bigger comment describing that maybe? Instead of explaining each row?

Reply:
Yes, a bigger comment might be better.

It feelse like we are getting a bit close to:
// incrementing i by 1
cnt += 1